examples/sldeditor/core/widgets/Segment.cpp

```diff
@@ -28,94 +30,68 @@ Segment::Segment(SegmentItem* item) :
     _right_handle(nullptr)
 {
     auto on_property_change = [this](ModelView::SessionItem*, std::string property_name) {
-        if (property_name == SegmentItem::P_X_FIN)
-            parentWidget()->update();
-        if (property_name == SegmentItem::P_X_INI)
-            parentWidget()->update();
-        if (property_name == SegmentItem::P_Y_INI)
-            parentWidget()->update();
-        if (property_name == SegmentItem::P_Y_FIN)
-            parentWidget()->update();
+        if (property_name == SegmentItem::P_X_POS)
+            setX(segment_item->property(SegmentItem::P_X_POS).toDouble());
+        if (property_name == SegmentItem::P_Y_POS)
+            setY(segment_item->property(SegmentItem::P_Y_POS).toDouble());
+        if (property_name == SegmentItem::P_HEIGHT)
+            update();
+        if (property_name == SegmentItem::P_WIDTH)
+            update();
+        if (property_name == SegmentItem::P_HORIZONTAL)
+            update();
         if (property_name == SegmentItem::P_COLOR)
-            parentWidget()->update();
+            update();
     };
     segment_item->mapper()->setOnPropertyChange(on_property_change, this);
 
-    double x_m = (
-        segment_item->property(SegmentItem::P_X_FIN).toDouble()
-        - segment_item->property(SegmentItem::P_X_INI).toDouble()) / 2
-        + segment_item->property(SegmentItem::P_X_INI).toDouble();
-    double y_m = (
-        segment_item->property(SegmentItem::P_Y_FIN).toDouble()
-        - segment_item->property(SegmentItem::P_Y_INI).toDouble()) / 2
-        + segment_item->property(SegmentItem::P_Y_INI).toDouble();
-
-    setPos(x_m,y_m);
+    setPos(
+        segment_item->property(SegmentItem::P_X_POS).toDouble(),
+        segment_item->property(SegmentItem::P_Y_POS).toDouble()
+    );
 
     setFlag(QGraphicsItem::ItemIsMovable);
 
 }
 
 void Segment::paint(QPainter* painter, const QStyleOptionGraphicsItem*, QWidget*)
 {
-    painter->setPen(QPen(color, 12));
-    painter->drawLine(
-        segment_item->property(SegmentItem::P_X_INI).toDouble(),
-        segment_item->property(SegmentItem::P_Y_INI).toDouble(),
-        segment_item->property(SegmentItem::P_X_FIN).toDouble(),
-        segment_item->property(SegmentItem::P_Y_FIN).toDouble());
+    painter->setBrush(color);
+    painter->drawRect(
+        - segment_item->property(SegmentItem::P_WIDTH).toDouble()/2,
+        - segment_item->property(SegmentItem::P_HEIGHT).toDouble()/2,
+        segment_item->property(SegmentItem::P_WIDTH).toDouble(),
+        segment_item->property(SegmentItem::P_HEIGHT).toDouble());
```

Comment:
It looks like `QPainter::drawRect(int x, int y, int w, int h)` overload is used here. I suggest to explicitly work with QPointF, QRectF in all painter's to avoid later problems with high-dpi.